pkg/objstore/gcs/gcs.go

```diff
@@ -88,72 +98,33 @@ func (b *Bucket) Handle() *storage.BucketHandle {
 	return b.bkt
 }
 
-// Exists checks if the given directory exists at the remote site (and contains at least one element).
-func (b *Bucket) Exists(ctx context.Context, dir string) (bool, error) {
-	b.opsTotal.WithLabelValues(opObjectsList).Inc()
-	objs := b.bkt.Objects(ctx, &storage.Query{
-		Delimiter: "/",
-		Prefix:    dir,
-	})
-	for {
-		_, err := objs.Next()
-		if err == iterator.Done {
-			return false, nil
-		}
-
-		if err != nil {
-			return false, err
-		}
+// Exists checks if the given object exists.
+func (b *Bucket) Exists(ctx context.Context, name string) (bool, error) {
+	b.opsTotal.WithLabelValues(opObjectGet).Inc()
 
-		// The first object found with the given filter indicates that the directory exists.
+	if _, err := b.bkt.Object(name).Attrs(ctx); err == nil {
```

Comment:
nice

Reply:
Yes, much better – just that callers need to specify some explicit file. But that seems saner anyway.